### ISSUE TITLE
espnet2 encountered better valid set effect,   but poor in test set.

### ISSUE THREAD

**Issue by Gavin90s** [espnet/espnet#3511]
**v1: wav2vec2.0 + ctc + att** 
[valid] loss=41.613, loss_att=33.179, loss_ctc=61.291, acc=0.753, cer=0.262, wer=nan, cer_ctc=0.328, time=1 minute and 25.13 seconds, total_count=109000, gpu_max_cached_mem_GB=9.686, [att_plot] time=11.07 seconds, total_count=0, gpu_max_cached_mem_GB=9.686

**baseline: ctc + att**
[valid] loss=34.135, loss_att=32.330, loss_ctc=38.348, acc=0.733, cer=0.315, wer=0.942, cer_ctc=0.346, time=16.61 seconds, total_count=4083, gpu_max_cached_mem_GB=8.717, [att_plot] time=32.88 seconds, total_count=0, gpu_max_cached_mem_GB=8.717

**language: zh, training set is 100h，valid set is 50h**

**$cat conf/decode_asr_rnn.yaml** 
beam_size: 2
penalty: 0.0
maxlenratio: 0.0
minlenratio: 0.0
ctc_weight: 0.6
lm_weight: 0.0

but test result with **valid.acc.best.pth** throught asr_inference.py script, the baseline cer is much better than v1. anyone can help me to find the reason of it ? any help well be greatly appreciated. @sw005320 @ShigekiKarita

**Comment by kan-bayashi** [comment]
Did you check the model average effect?

**Comment by Gavin90s** [comment]
> Did you check the model average effect?

i did not check model average effect.

**Comment by kan-bayashi** [comment]
As a default, averaged model will be used for the inference. Maybe the comparison results using train.acc.best.pth gives more insights.

**Comment by Gavin90s** [comment]
> As a default, averaged model will be used for the inference. Maybe the comparison among results using train.acc.best.pth gives more insights.

thanks for your reply. i will try it immediately.

**Comment by Gavin90s** [comment]
> Although the validation accuracy looks good, the loss (especially CTC loss) is very bad.
> I think the training does not go well in the encoder, and the decoder is simply over-trained.
> I recommend you tune some optimizer related hyper-parameters.

the **valid** ctc loss of wav2vec2.0 encoder seems to converge well, but always higher than that of conformer encoder, when the **valid ctc_cer** and **training ctc loss** looks better.
